### PR TITLE
docs: fix typos in prebuilt-binaries.md and release-guide.md

### PR DESCRIPTION
**Pull request by 0xbryer**
## Overview

**Description**:
This pull request addresses two spelling errors in the documentation files:

1. **In `docs/maintainers/prebuilt-binaries.md`**:
   - The word **occured** is corrected to **occurred** (with two "r"s).
   
2. **In `docs/maintainers/release-guide.md`**:
   - The word **relase** is corrected to **release** (with an extra "e").

**Importance**:
These spelling corrections ensure that the documentation is accurate and professional. Correct spelling is important in maintaining clarity and trust in the documentation, especially for new contributors and users. It also improves searchability, as users may encounter issues when searching for these terms with the incorrect spelling.

### CODE REVIEW

**Comment by 0xbryer** [comment]
[rootulp](https://github.com/rootulp) please merge it.

**Comment by rootulp** [comment]
This needs two approvals. After the second approval, it will auto-merge.

**Review by rach-id — APPROVED** [review]

**Pull request merged**
